src/VuMeter.cpp

```diff
@@ -0,0 +1,155 @@
+  The above copyright notice and this permission notice shall be
+  included in all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+  OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+  WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+  OTHER DEALINGS IN THE SOFTWARE.
+*/
+//*****************************************************************
+
+/**
+ * \file Limiter.cpp
+ * \author Dominick Hing
+ * \date August 2022
+ * \license MIT
+ */
+
+#include "VuMeter.h"
+
+#include <QVector>
+
+#include "jacktrip_types.h"
+
+//*******************************************************************************
+void VuMeter::init(int samplingRate)
+{
+    ProcessPlugin::init(samplingRate);
+    if (samplingRate != fSamplingFreq) {
+        std::cerr << "Sampling rate not set by superclass!\n";
+        std::exit(1);
+    }
+
+    fs = float(fSamplingFreq);
+    for (int i = 0; i < mNumChannels; i++) {
+        vumeterP[i]->init(fs);
+        // int ndx = vumeterUIP[i]->getParamIndex("NumClientsAssumed");
+        // vumeterUIP[i]->setParamValue(ndx, mNumClients);
+    }
+
+    /* Set meter values to the default floor */
+    mValues.resize(mNumChannels);
+    QVector<float>::iterator it;
+    for (it = mValues.begin(); it != mValues.end(); ++it) {
+        *it = -80.0;
+    }
+
+    /* Start timer */
+    int timeout_ms = 100;
+    connect(&mTimer, &QTimer::timeout, this, &VuMeter::onTick);
+    mTimer.setTimerType(Qt::PreciseTimer);
+    mTimer.setInterval(timeout_ms);
+    mTimer.setSingleShot(false);
+    mTimer.start();
+
+    inited = true;
+}
+
+//*******************************************************************************
+void VuMeter::compute(int nframes, float** inputs, float** /*_*/)
+{
+    // Note that the second parameter is unused. This is because all of the ProcessPlugins
+    // require the same function signature for the compute() function and is normally used
+    // for the faust plugin output. However, this plugin is not supposed to modify the
+    // signal itself like the other plugins (e.g. Limiter) do, so we don't want to write
+    // to this buffer. We just need to report the VU meter output
+
+    if (not inited) {
+        std::cerr << "*** VuMeter " << this << ": init never called! Doing it now.\n";
+        if (fSamplingFreq <= 0) {
+            fSamplingFreq = 48000;
+            std::cout << "Limiter " << this
+                      << ": *** HAD TO GUESS the sampling rate (chose 48000 Hz) ***\n";
+        }
+        init(fSamplingFreq);
+    }
+
+    for (int i = 0; i < mNumChannels; i++) {
+        /* Run the signal through Faust to  */
+        QVector<float> meterBuf(nframes);
```

Comment:
given that `nframes` does not change, this (and the next two `float` statements) can be moved outside of the `for` loop.

Reply:
done